editor/src/main/res/values/attrs.xml

```diff
@@ -26,34 +26,127 @@
+<!--        <attr name="lineNumberVisible" format="boolean" />-->
+<!--        <attr name="autoCompleteEnabled" format="boolean" />-->
+<!--        <attr name="symbolCompletionEnabled" format="boolean" />-->
+<!--        <attr name="cursorBlinkPeriod" format="integer" />-->
+<!--        <attr name="scrollbarsEnabled" format="boolean" />-->
+<!--        <attr name="verticalScrollbarEnabled" format="boolean" />-->
+<!--        <attr name="horizontalScrollbarEnabled" format="boolean" />-->
+<!--    </declare-styleable>-->
+
     <declare-styleable name="CodeEditor">
-        <attr name="android:scrollbarTrackHorizontal" />
-        <attr name="android:scrollbarThumbHorizontal" />
-        <attr name="android:scrollbarTrackVertical" />
-        <attr name="android:scrollbarThumbVertical" />
-
-        <attr name="lnPanelPositionMode">
-            <enum name="fixed" value="0" />
-            <enum name="follow" value="1" />
+
+        <!-- General -->
+        <attr name="editorColorScheme">
+            <enum name="normal" value="0" />
+            <enum name="darcula" value="1" />
+            <enum name="eclipse" value="2" />
+            <enum name="github" value="3" />
+            <enum name="notepadXX" value="4" />
+            <enum name="vs2019" value="5" />
+        </attr>
+        <attr name="cursorBlinkPeriod" format="integer" />
+        <attr name="cursorWidth" format="dimension" />
+
+        <attr name="displayLinePanel" format="boolean" />
+        <attr name="dividerWidth" format="dimension" />
+        <attr name="dividerMarginLeft" format="dimension"/>
+        <attr name="dividerMarginRight" format="dimension"/>
+
+        <attr name="editable" format="boolean" />
+        <attr name="edgeEffectColor" format="color" />
+
+        <attr name="highlightBracketPair" format="boolean" />
+
+        <attr name="ligatureEnabled" format="boolean" />
+        <attr name="lineNumberAlign">
+            <enum name="left" value="6" />
+            <enum name="right" value="7" />
+            <enum name="center" value="8" />
+        </attr>
+        <attr name="lineNumberEnabled" format="boolean" />
+        <attr name="lineNumberMarginLeft" format="dimension" />
+        <attr name="lineSpacingExtra" format="dimension" />
+        <attr name="linePanelInfoTextSize" format="dimension" />
+        <attr name="linePanelPositionMode">
+            <enum name="fixed" value="9"/>
+            <enum name="follow" value="10"/>
         </attr>
-        <attr name="lnPanelPosition">
-            <flag name="left" value="1" />
-            <flag name="top" value="2" />
-            <flag name="right" value="4" />
-            <flag name="bottom" value="8" />
-            <flag name="center" value="15" />
+        <attr name="linePanelPosition">
+            <flag name="left" value="11"/>
+            <flag name="top" value="12"/>
+            <flag name="right" value="13"/>
+            <flag name="bottom" value="14"/>
```

Comment:
the value of of left/top/right/bottom should be restored because these values can be XORd to indicate left-top and other locations.